src/imperative/cached_op.h

```diff
@@ -294,7 +294,7 @@ void SetInputIndices(const nnvm::Graph& fwd_graph,
   const auto& indexed_graph = fwd_graph.indexed_graph();
   if (data_indices->ndim() || param_indices.ndim()) {
     CHECK_EQ(data_indices->ndim() + param_indices.ndim(),
-             indexed_graph.input_nodes().size());
+             static_cast<const int>(indexed_graph.input_nodes().size()));
```

Comment:
Is this what fixed that weird error
```
[2020-02-05T20:57:05.668Z] /work/mxnet/3rdparty/mshadow/mshadow/././logging.h:75:34: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
[2020-02-05T20:57:05.668Z]  #define CHECK_EQ(x, y) CHECK((x) == (y))
[2020-02-05T20:57:05.668Z]                                   ^
[2020-02-05T20:57:05.668Z] /work/mxnet/3rdparty/mshadow/mshadow/././logging.h:68:9: note: in definition of macro 'CHECK'
[2020-02-05T20:57:05.668Z]    if (!(x))                                                \
[2020-02-05T20:57:05.668Z]          ^
[2020-02-05T20:57:05.668Z] src/imperative/./cached_op.h:290:5: note: in expansion of macro 'CHECK_EQ'
[2020-02-05T20:57:05.668Z]      CHECK_EQ(data_indices->ndim() + param_indices.ndim(),
[2020-02-05T20:57:05.668Z]      ^
[2020-02-05T20:57:05.668Z] cc1plus: all warnings being treated as errors
```

Reply:
Yes. Unfortunately there seems to be further undefined behavior and the CI needs more fixing (see failure status below.. Can't reproduce any of these failures locally..)

Reply:
Was able to reproduce with Makefile based build and older gcc. With that setup, only 

``` c++
#include "./imperative_utils.h"
#include "./cached_op.h"
```

works in `imperative.cc`, but not

``` c++
#include "./cached_op.h"
#include "./imperative_utils.h"
```

which was used in the PR initially.